extensions/amp-youtube/0.1/amp-youtube.js

```diff
@@ -71,6 +84,48 @@ class AmpYoutube extends AMP.BaseElement {
     // the playback.
     return false;
   }
+
+  /** @private */
+  buildImagePlaceholder_() {
+    const imgPlaceholder = new Image();
+    const videoid = this.videoid_;
+    function loadHQImage_() {
+      // We only need to handle error once to load another source.
+      imgPlaceholder.removeEventListener('error', loadHQImage_);
+      imgPlaceholder.setAttribute('src', 'https://i.ytimg.com/vi/' +
+          videoid + '/hqdefault.jpg');
+    }
+
+    function maybeLoadHQImage_() {
+      // We only need to handle error once to load another source.
+      imgPlaceholder.removeEventListener('load', maybeLoadHQImage_);
+      // A pretty ugly hack since onerror won't fire on YouTube image 404.
+      // This might be due to the fact that YouTube returns data to the request
+      // even when the status is 404.
+      if ((this.naturalWidth === 120 && this.naturalHeight === 90) ||
+          (this.width === 120 && this.height === 90)) {
+        loadHQImage_();
+      } else {
+        imgPlaceholder.removeEventListener('error', loadHQImage_);
+      }
+    }
+
+    // Because sddefault.jpg isn't available for all videos, we try to load
+    // it and fallback to hqdefault.jpg.
+    imgPlaceholder.addEventListener('load', maybeLoadHQImage_);
+    imgPlaceholder.addEventListener('error', loadHQImage_);
+
+    // TODO(mkhatib): Maybe add <source> children to allow the browser to
+    // load the needed size or even better match YTPlayer logic for loading
+    // player thumbnails for different screen sizes for a cache win!
+    imgPlaceholder.setAttribute('src', 'https://i.ytimg.com/vi/' +
+        this.videoid_ + '/sddefault.jpg');
+    imgPlaceholder.setAttribute('placeholder', '');
+    imgPlaceholder.width = this.width_;
+    imgPlaceholder.height = this.height_;
+    this.element.appendChild(imgPlaceholder);
+    this.applyFillContent(imgPlaceholder);
```

Comment:
I think image placeholder will be fine - it will be hidden automatically. It's just always good to know when the major work has been complete rather than guessing based on the iframe.onload, which usually means very little.